### PR TITLE
[IOTDB-3758] support merge in new cluster

### PR DESCRIPTION
**Pull request by RYH61**
MERGE (ON (LOCAL | CLUSTER))?
FULL MERGE (ON (LOCAL | CLUSTER))?

In PlanExecutor.java
图片

So MERGE and FULL MERGE temporary use same method

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java [hunk 1]
```diff
@@ -2325,6 +2314,36 @@ private Pair<Long, Long> calcOperatorInterval(QueryFilter queryFilter) {
     }
   }
 
+  // Merge
+
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java:2318]: remove space.

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java [hunk 2]
```diff
@@ -483,9 +485,19 @@ public TSStatus invalidatePermissionCache(TInvalidatePermissionCacheReq req) {
     return RpcUtils.getStatus(TSStatusCode.INVALIDATE_PERMISSION_CACHE_ERROR);
   }
 
+  @Override
+  public TSStatus merge(TMergeReq req) throws TException {
+    try {
+      storageEngine.mergeAll();
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:491]: why merge all ?  req does noet used?

**Comment by wangchao316** [comment]
add IT

**Review by Beyyes — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java (hunk 1 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java]: fix~

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java (hunk 2 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java:489]: It's not used here. I've removed it

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**